utils/update_dns.sh

```diff
@@ -50,7 +50,7 @@ function update_record() {
 
 echo -n "Checking local IP Address: "
 my_ip=$(curl -s ifconfig.me)
-my_ipv6=$(ip addr show scope global |grep inet6 |head -1|perl -n -e'/inet6\s([0-9a-f:]+)\/\d+\s/&& print $1')
+my_ipv6=$(ip addr show scope global |grep inet6 |grep -v fe80 |grep -v 'inet6 fd'|head -1|perl -n -e'/inet6\s([0-9a-f:]+)\/\d+\s/&& print $1')
```

Comment:
ip -6 addr | grep inet6 | awk -F '[ \t]+|/' '{print $3}' | grep -v ^::1 | grep -v ^fe80

https://superuser.com/a/1057290
maybe slightly better

Reply:
On my system this gives me back 4 IPs. 2 are local (the fd34, is part of the reserved for local range fc* and fd*) https://en.wikipedia.org/wiki/Unique_local_address

```
[~]$ ip -6 addr | grep inet6 | awk -F '[ \t]+|/' '{print $3}' | grep -v ^::1 | grep -v ^fe80
fd34:9d98:a2d2:1::433
2601:645:****:****::433
fd34:9d98:a2d2:1:de84:c7ad:39ea:dddf
2601:645:****:****:235e:b65b:17f9:9328
```
On my setup (Xfinity) the /128 address seems to be semi permanent (changes only when my prefix changes every N months and the suffix never changes) and the 2601 with a /64 mask is a temporary address which changes fairly often so probably not a good candidate to serve the website reliably

I could go with complicated regex but I feel like it's probably easier to read if I do:
```
ip -6 addr show scope global \
  |grep -v 'inet6 fe80' \
  |grep -v 'inet6 fc' \
  |grep -v 'inet6 fd' \
  |perl -lne'/inet6\s([0-9a-f:]+)\/\d+/ && print $1' \
  |head -1
```

This works well on 3 different ISP. On Xfinity, I luck out because the first /128 is selected. Can you test it too?

Reply:
Whoa interesting Xfinity behavior.
On mine, the first one (that `head -n1` gets) is the ipv6 privacy address on my link.
Servers though probably won't have the ipv6 privacy stuff turned on?
There may not be a "best" way to do it. :shrug: